File: js/026_RemoveDuplicatesfromSortedArray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
    if(nums.length === 0) return 0;
    var j = 0;
    for( var i = 0; i < nums.length; ++i){
        if(nums[i] !== nums[j]){
            nums[j+1] = nums[i];
            ++j;
        }
    }
    return j + 1;
};

File: cpp/026_RemoveDuplicatesfromSortedArray.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if(nums.size() == 0) return 0;
        int j = 0;
        for( int i = 1; i < nums.size(); ++i ){
            if(nums[i] != nums[j]){
                nums[j+1] = nums[i];
                ++j;
            }
        }
        return j+1;
    }
};